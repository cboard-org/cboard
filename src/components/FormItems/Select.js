import React from 'react';
import { InputLabel } from 'material-ui/Input';
import { MenuItem } from 'material-ui/Menu';
import { FormControl, FormHelperText } from 'material-ui/Form';
import MUISelect from 'material-ui/Select';

const Select = ({ error, label, options, ...props }) => (
  <FormControl className="SignUp__field" error={!!error}>
    <InputLabel>{label}</InputLabel>
    <MUISelect {...props}>
      {options.map(opt => (
        <MenuItem key={opt.value} value={opt.value}>
          {opt.label}
        </MenuItem>
      ))}
    </MUISelect>
    {error && <FormHelperText>{error}</FormHelperText>}
  </FormControl>
);

export default Select;
