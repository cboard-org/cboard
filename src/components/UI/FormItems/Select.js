import React from 'react';
import PropTypes from 'prop-types';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import MUISelect from '@material-ui/core/Select';

const propTypes = {
  className: PropTypes.string,
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number
  ]),
  label: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired
};

const defaultProps = {
  className: '',
  error: false,
  label: ''
};

const Select = ({ className, error, label, options, ...props }) => (
  <FormControl className={className} error={!!error}>
    {label && <InputLabel>{label}</InputLabel>}
    <MUISelect {...props}>
      {options.map(opt => (
        <MenuItem key={opt.value} value={opt.value}>
          {opt.label}
        </MenuItem>
      ))}
    </MUISelect>
    {error && <FormHelperText>{error}</FormHelperText>}
  </FormControl>
);

Select.propTypes = propTypes;
Select.defaultProps = defaultProps;

export default Select;
