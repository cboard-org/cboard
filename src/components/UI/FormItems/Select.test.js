import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import Select from './Select';

describe('Select tests', () => {
  test('default renderer', () => {
    const options = [
      { value: 'v1', label: 'l1' },
      { value: 'v2', label: 'l2' },
      { value: 'v3', label: 'l3' }
    ];
    shallowMatchSnapshot(<Select options={options} />);
  });
});
