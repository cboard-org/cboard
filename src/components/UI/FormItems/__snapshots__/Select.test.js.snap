// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select tests default renderer 1`] = `
<WithStyles(FormControl)
  className=""
  error={false}
>
  
  <WithStyles(Select)>
    <WithStyles(MenuItem)
      value="v1"
    >
      l1
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      value="v2"
    >
      l2
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      value="v3"
    >
      l3
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</WithStyles(FormControl)>
`;
